### DISCUSSION TITLE
Is is possible to de-compile SQL into PRQL ?

### DISCUSSION THREAD

**Discussion by xulai1001** [PRQL/prql#4684] [General]
AFAIK, PRQL compiles to different SQL dialects easily with prqlc.
Is is possible to decompile existing SQL queries into PRQL format, something like "sql_to_pl", so that I can convert between different sql dialects with prql?

**Comment by xulai1001** [comment]
BTW is it possible to generate insert/update statements.....?

**Reply by vanillajonathan** [reply]
No, PRQL only does querying. There has been talks about mutation queries in #1092.

**Comment by vanillajonathan** [comment]
No, PRQL does not support yet SQL-to-PRQL conversion but it is mentioned in the roadmap as a long-term goal. https://prql-lang.org/roadmap/

**Reply by xulai1001** [reply]
Oooooookay.... Thanks for your detailed answer!

**Comment by ukclivecox** [comment]
Any update or further work on SQL-to-PRQL?

**Reply by vanillajonathan** [reply]
No, as far as I know there have been no work on that and considering all the SQL dialects it sounds like something that would be difficult.
